### PR TITLE
Modified JSDocs notation in each function and create JSdocs server to…

### PR DESCRIPTION
**Pull request by ademiltonnunes**
… run the documentation

# Description
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Ticket: Backend - Modify API Documentation

## Type of change
Please delete options that are not relevant.
- [X] Change Documentation for Firebase Functions

### CODE REVIEW

**Review by abhi-s7 — APPROVED** [review]
I reviewed the changes and I am good with them. 
You can go ahead and push to the main branch.

File: functions/controllers/kaiAIController.js [hunk 1]
```diff
@@ -12,21 +12,20 @@ const app = express();
 
 const DEBUG = process.env.DEBUG;
 /**
- * Simulates communication with a Kai AI endpoint.
+ * Communicates with a Kai AI endpoint.
```
Comment (sivaShilpa) [inline functions/controllers/kaiAIController.js]: Please return this back as "Simulates communication" as it makes more sense.

File: functions/controllers/kaiAIController.js [hunk 2]
```diff
@@ -12,21 +12,20 @@ const app = express();
...
+ * @param {Object} payload - The properties of the communication.
+ * @param {Object} payload.data - The payload data object used in the communication.
+ * @param {Array} [payload.data.messages] - An array of messages for the current user chat session.
+ * @param {Object} payload.data.user - The user object.
+ * @param {Object} [payload.data.tool_data] - The tool-specific data object.
+ * @param {string} payload.data.type - The type of communication (e.g., 'chat', 'tool').
+ * @returns {Promise<Object>} A promise that resolves to an object with status and data properties.
+ * @returns {string} return.status - The status of the operation ('success').
+ * @returns {Object} return.data - The data returned from the AI service.
+ * @throws {HttpsError} If there's an error in communication or processing.
```
Comment (sivaShilpa) [inline functions/controllers/kaiAIController.js]: I would refrain from editing the docs that were already there as they may have had different meanings than what you are trying to mean with your edits. Let Billy go through it and confirm this.

File: functions/controllers/kaiAIController.js [hunk 3]
```diff
@@ -266,18 +269,21 @@ app.post('/api/tool/', (req, res) => {
...
- * This creates a chat session for a user.
+ * Creates a chat session for a user.
  * If the chat session already exists, it will return the existing chat session.
  * Otherwise, it will create a new chat session and send the first message.
  *
+ * @async
+ * @function createChatSession
  * @param {Object} props - The properties passed to the function.
- * @param {Object} props.data - The data object containing the user, challenge, message, and botType.
+ * @param {Object} props.data - The data object containing the user, message, and type.
```
Comment (sivaShilpa) [inline functions/controllers/kaiAIController.js]: The previous doc was more clear than this. Please put it back.

File: functions/controllers/kaiAIController.js [hunk 4]
```diff
@@ -266,18 +269,21 @@ app.post('/api/tool/', (req, res) => {
 });
 
 /**
- * This creates a chat session for a user.
+ * Creates a chat session for a user.
```
Comment (sivaShilpa) [inline functions/controllers/kaiAIController.js]: Not sure why you had to correct this.

**Review by sivaShilpa — COMMENTED** [review]
Great work Marcelo! But I would refrain from editing things that were already there unless it is really important for our current expedition. Please take notes of what I have commented and review them. Make changes only after Billy has reviewed it. 

Thank you!

**Comment by ademiltonnunes** [comment]
Sorry, I didn't include many details in my PR description.

**Description**
This pull request updates the API documentation for Firebase Functions, specifically focusing on the Kai Platform and user management functionalities. The changes include:

1. Implementation of JSDoc comments in the codebase to generate comprehensive API documentation.
2. Addition of a README file with instructions on how to generate, view, and maintain the documentation.

These changes aim to improve the overall developer experience by providing clear, up-to-date documentation of our Firebase Functions. This will facilitate frontend team and serve as a quick reference for existing developers.

**Motivation:** The previous documentation was outdated and not running. This update addresses this  issue.

**Dependencies:** This change requires the installation of JSDoc and http-server as dev dependencies in the project.

**Ticket:** Backend - Modify API Documentation

**Type of change**
* Change Documentation for Firebase Functions

File: functions/controllers/kaiAIController.js (hunk 3 above)
Comment (ademiltonnunes) [reply functions/controllers/kaiAIController.js]: modified as your suggestions

File: functions/controllers/kaiAIController.js (hunk 4 above)
Comment (ademiltonnunes) [reply functions/controllers/kaiAIController.js]: it was generated by the toll I was using. Undone as your suggestion.

File: functions/controllers/kaiAIController.js (hunk 2 above)
Comment (ademiltonnunes) [reply functions/controllers/kaiAIController.js]: Undo modification

**Review by sivaShilpa — APPROVED** [review]
Good Work!

**Pull request merged**